maps/mp/isnipe/extras/peoplealive.gsc

```diff
@@ -1,78 +1,72 @@
+ */
+init() {
+    level.peopleAlive = [];
+    level.peopleAlive["allies"] = 0;
+    level.peopleAlive["axis"] = 0;
 }
 
-
-
-self.ateam = self.pers["team"];
-
-if (self.ateam == "allies")
-    {
-    level.alliesalive += 1;
-    } else if (self.ateam == "axis") {
-    level.axisalive += 1;
-    }
-
-self thread DoLabel();
-
-self waittill("death");
-
-if (self.ateam == "allies")
-    {
-    level.alliesalive -= 1;
-    } else if (self.ateam == "axis") {
-    level.axisalive -= 1;
-    }
-
+/**
+ * Initialize the variables used by the peopleAlive module on a player level.
+ */
+onPlayerConnected() {
+    self.peopleAlive = [];
+    self.peopleAlive["hud"] = [];
 }
 
-DoLabel()
-{
-self endon("death");
-lbl = self createFontString("hudbig", 0.8);
-lbl setPoint("TOPLEFT", "TOPLEFT", 44, 110);
-self thread deleteondeath(lbl);
-lbl.hideWhenInMenu = true;
-for(;;)
-{
-
-if (self.ateam == "allies")
-    {
-    if (getDvar("g_gametype") != "dm"){
-
-lbl setText("^2" + level.alliesalive+ " ^1" +level.axisalive);
-    } else {
-    //FREE FOR ALL
-
-    lbl setText("^21" + " ^1" + (level.players.size -1));
-
-    }
-    } else if (self.ateam == "axis") {
-        if (getDvar("g_gametype") != "dm"){
-lbl setText("^2" + level.axisalive+ " ^1" + level.alliesalive);
-} else {
-lbl setText("^21" + " ^1" + (level.players.size - 1));
+/**
+ * Log player alive status as well as drawing the HUD for the player.
+ */
+onPlayerSpawned() {
+    team = self.pers["team"];
+    level.peopleAlive[team]++;
+    self thread _drawHUD();
+    self thread _deleteHUD();
+    level notify("peopleAlive:UPDATE");
+
+    self waittill("death");
+    level.peopleAlive[team]--;
+    level notify("peopleAlive:UPDATE");
 }
+
+/**
+ * Draw and update the player HUD on alive people.
+ */
+_drawHUD() {
+    self endon("death");
+    self endon("disconnect");
+
+    label = self createFontString("hudbig", 0.8);
+    label setPoint("TOPLEFT", "TOPLEFT", 44, 110);
+    label.hideWhenInMenu = true;
+    self.peopleAlive["hud"][0] = label;
+    for (;;) {
+        // Check whether gamemode is FFA.
+        if (getDvar("g_gametype") == "dm") {
+            // TODO: Does "level.players.size" give all connected players or alive players?
+            label setText("^21 ^1" + (level.players.size - 1));
+        } else {
+            if (self.pers["team"] == "allies") {
+                label setText("^2" + level.peopleAlive["allies"] + " ^1" + level.peopleAlive["axis"]);
+            } else {
+                label setText("^2" + level.peopleAlive["axis"] + " ^1" + level.peopleAlive["allies"]);
+            }
+        }
+        level waittill("peopleAlive:UPDATE");
     }
-wait 0.1;
-}
 
 }
 
-deleteondeath(hud)
-{
-        self waittill("death");
-        hud destroy();
+/**
+ * Delete the HUD after the player has died.
+ */
+_deleteHUD() {
+    self waittill("death");
+    foreach (hudElement in self.peopleAlive["hud"]) {
+        hudElement.destroy();
```

Comment:
This throws:
 `******* script compile error *******
Error: bad syntax in file maps/mp/isnipe/extras/peoplealive.gsc, line 69:
        hudElement.destroy();
                         *
************************************`

It should be `hudElement destroy();`